### ISSUE TITLE
[miopen-hip] Can't download boost 1.72.0 dependency because bintray.com shut down on May 1st 2021

### ISSUE THREAD

**Issue by darjankrijan** [rocm-arch/rocm-arch#596]
During fetching some dependencies at the start of the build, this error occurs:

```
Downloading https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2
-- [download 0% complete]
CMake Error at /build_root/miopen-hip/src/deps/share/cmake/cmakeget/CMakeGet.cmake:151 (message):
  Download failed: "HTTP response code said error":
  https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2;/tmp/cget-16-39-03-XLN6p-1/download/boost_1_72_0.tar.bz2;SHOW_PROGRESS
```

Upstream issue that apparently resolved this in another repo pointed to by install_deps.cmake in miopen's root:
https://github.com/ROCmSoftwarePlatform/MIOpen/issues/856
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/install_deps.cmake#L96

Not sure exactly what is happening and how to patch this.

**Comment by tpkessler** [comment]
That's weird. I don't really understand cmake scripts but as far as I see, `miopen-{hip, opencl}` should pull the URLs from https://github.com/pfultz2/rocm-recipes. A quick fix is to replace the line `boost@1.72` in `min-requirements.txt` with
```
boost,https://sourceforge.net/projects/boost/files/boost/1.72.0/boost_1_72_0.tar.bz2 -H sha256:59c9b274bc451cf91a9ba1dd2c7fdcaf5d60b1b3aa83f2c9fa143417cc660722 -X boost -DBOOST_WITHOUT_CONTEXT=1 -DBOOST_WITHOUT_COROUTINE=1 -DBOOST_WITHOUT_PYTHON=1
```

**Comment by tpkessler** [comment]
Patched with 1992510.

**Comment by tpkessler** [comment]
With version 4.3.0, boost is downloaded from sourceforge by default.
